### PR TITLE
Implement Instantiate2

### LINKED ISSUE #12: Implement Instantiate2
There should be a way to predictably instantiate contracts with localic-utils. This should be done with instantiate2, and should follow the existing builder pattern for write operations. Furthermore, a query method should be added for build-contract-address to get the generated address.

## Suggested Changes

- Implement an instantiate2 tx builder / sender
- Implement a query for build-address

### PR DESCRIPTION
**Pull request by lexzaiello**
Closes #12 . There is currently no way to predictably instantiate a contract with localic-std. This PR adds support for the instantiate2 CLI command via a new builder: `build_tx_instantiate2`. Some other housekeeping items were addressed, enumerated below.

## Overview of Changes

- Renamed `fixtures.rs` to `queries.rs` to be more intuitive and accurate
- Removed some unnecessary cloning and cleaned up existing TestContextQuery query impls
- Implemented a builder and `.send` method for instantiate2
- Implemented a query for getting a code ID's info (`q code-info`)
- Implemented a query for getting an instantiate2 address (`.get_built_contract_address`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,6 +97,17 @@ Note that most `tx_*` helper functions expose a `.with_key(key: &str)` builder f
...
+#### Wasm
+
+* `.build_tx_instantiate2` - Predictably instantiates a CosmWasm contract.
+  * Required builder calls:
+    * `.with_code_id(code_id: u64)` - Should be the raw code ID of the contract being instantiated
+	* `.with_salt_hex_encoded(salt: &str)` - Should be a **hex-encoded** salt for instantiation
+	* `.with_msg(msg: serde_json::Value)`
+	* `.with_label(label: &str)`
+  * Notable optional builder calls:
+    * `.with_chain_name(chain_name: impl Into<String>)` - Should be on of `"osmosis" | "neutron" | "stride"` or one of the registered chain names from `.with_chain`
```
Comment (keyleu) [inline README.md]: typo: one*. Stride doesn't have cosmwasm so we can remove it from this list.

File: src/utils/test_context.rs [hunk 2]
```diff
@@ -531,110 +559,123 @@ impl<'a> TestContextQuery<'a> {
...
-                .chains
-                .get(src)
-                .map(|chain| chain.chain_prefix.clone())
-        } else {
-            None
+    fn get_built_contract_address(&self) -> Option<String> {
+        let code_info = self.get_code_info()?;
+        let code_id_hash = code_info["data_hash"].as_str()?;
+
+        let creator_address = NEUTRON_CHAIN_ADMIN_ADDR;
```
Comment (keyleu) [inline src/utils/test_context.rs]: this should be self.creator_address

**Review by keyleu — CHANGES_REQUESTED** [review]

File: src/utils/test_context.rs [hunk 3]
```diff
@@ -531,110 +559,123 @@ impl<'a> TestContextQuery<'a> {
...
+        let salt = self.salt_hex_encoded.as_deref()?;
+
+        let chain = self
+            .context
+            .chains
+            .get(self.src_chain.as_deref().unwrap_or(NEUTRON_CHAIN_NAME))?;
+
+        // text field contains built address
```
Comment (keyleu) [inline src/utils/test_context.rs]: can we move this comment to the line with the match

**Comment by keyleu** [comment]
Nice refactor of the queries! Only some small changes required and looks good

File: examples/neutron.rs [hunk 4]
```diff
@@ -124,5 +126,40 @@ fn main() -> Result<(), Box<dyn Error>> {
         .with_liq_token_receiver(ACC_0_ADDR)
         .send()?;
 
+    let factory_contract_code_id = ctx
+        .get_contract("astroport_whitelist")
+        .unwrap()
+        .code_id
+        .unwrap();
+
+    // Instantiate a contract predictably
```
Comment (keyleu) [inline examples/neutron.rs]: very minor nit: can we rephrase to // Instantiate a contract with a predictable address

File: README.md (hunk 1 above)
Comment (lexzaiello) [reply README.md]: Fixed!

File: src/utils/test_context.rs (hunk 2 above)
Comment (lexzaiello) [reply src/utils/test_context.rs]: Fixed!

File: src/utils/test_context.rs (hunk 3 above)
Comment (lexzaiello) [reply src/utils/test_context.rs]: Done!

File: examples/neutron.rs (hunk 4 above)
Comment (lexzaiello) [reply examples/neutron.rs]: Done!

File: src/utils/test_context.rs (hunk 2 above)
Comment (keyleu) [reply src/utils/test_context.rs]: I think you forgot to push the changes

Comment (lexzaiello) [reply src/utils/test_context.rs]: Oh whoops

Comment (lexzaiello) [reply src/utils/test_context.rs]: :skull: I was on my TestContextQuery refactor branch

File: README.md (hunk 1 above)
Comment (keyleu) [reply README.md]: typo is still there: on -> one*

Comment (lexzaiello) [reply README.md]: Fixed!

**Review by keyleu — APPROVED** [review]

**Pull request merged**